### PR TITLE
Update Tests, Remove a lots of guava calls

### PR DESCRIPTION
**Pull request by kadir1243**
This pull contains:
- Remove a lots of guava calls(It Should decrease size of shadowed jars)
- Update Tests to Junit 5
- Change Some Throw Statements (I hope this will not broke anything)
- Update dependencies
- Change Local Variable Name Fixer system to detect compiled java version and not apply fixes if not needed for java version

### CODE REVIEW

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java [hunk 1]
```diff
@@ -111,19 +110,17 @@ private EntryTree<EntryMapping> read(Path path, List<String> lines, ProgressList
 
 						if (state.get(IN_CLASS)) {
 							switch (parts[0]) {
-								case "m": // method
+								case "m" -> { // method
 									state.set(IN_METHOD);
 									holds[IN_METHOD] = parseMethod(holds[IN_CLASS], parts, escapeNames);
-									break;
```
Comment (NoComment1105) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java:117]: Why have all the breaks been removed from this switch? Have you verified they're unnecessary?

File: enigma/src/main/java/cuchaz/enigma/translation/representation/TypeDescriptor.java [hunk 2]
```diff
@@ -43,7 +42,7 @@ public TypeDescriptor(String desc) {
 
 	public static String parseFirst(String in) {
 
-		if (in == null || in.length() <= 0) {
+		if (in == null || in.length() == 0) {
```
Comment (NoComment1105) [inline enigma/src/main/java/cuchaz/enigma/translation/representation/TypeDescriptor.java:45]: Have you verified this and other locations don't carry the risk of being less than 0?

**Review by NoComment1105 — CHANGES_REQUESTED** [review]
Looking mostly good, there's a lot of inconsistency around removal of some wildcard imports but conversion to wildcard imports that could do with being looked at if possible. I'm not an Enigma dev but I like what I see :P

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java (hunk 1 above)
Comment (Geolykt) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java:117]: I believe they are not required with the new switch-case syntax

Comment (NoComment1105) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java:117]: Interesting, I'm not convinced

File: enigma/src/main/java/cuchaz/enigma/translation/representation/TypeDescriptor.java (hunk 2 above)
Comment (Geolykt) [reply enigma/src/main/java/cuchaz/enigma/translation/representation/TypeDescriptor.java:45]: Have you ever seen a string that is -1 codepoints long? It's fair to assume that in the event of there being an integer overflow that we have larger worries (if we were to assume that the JVM supports arrays of that size - which I don't think it does)

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/srg/SrgMappingsWriter.java [hunk 3]
```diff
@@ -110,8 +110,8 @@ private String describeField(FieldEntry entry) {
 		return entry.getParent().getFullName() + "/" + entry.getName();
 	}
 
-	private Collection<Entry<?>> sorted(Iterable<? extends Entry<?>> iterable) {
-		ArrayList<Entry<?>> sorted = Lists.newArrayList(iterable);
+	private Collection<Entry<?>> sorted(Collection<? extends Entry<?>> iterable) {
+		List<Entry<?>> sorted = new ArrayList<>(iterable);
```
Comment (ByMartrixx) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/srg/SrgMappingsWriter.java:114]:
```suggestion
	private Collection<Entry<?>> sorted(Collection<? extends Entry<?>> entries) {
		List<Entry<?>> sorted = new ArrayList<>(entries);
```

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java (hunk 1 above)
Comment (ByMartrixx) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/tinyv2/TinyV2Reader.java:117]: Yeah, switch expressions don't need breaks

File: enigma/src/main/java/cuchaz/enigma/utils/Utils.java [hunk 4]
```diff
@@ -51,6 +43,19 @@ public static void delete(Path path) throws IOException {
...
+        Path name = path.getFileName();
+
+        // null for empty paths and root-only paths
+        if (name == null) {
+            return "";
+        }
+
+        String fileName = name.toString();
+        int dotIndex = fileName.lastIndexOf('.');
+        return dotIndex == -1 ? "" : fileName.substring(dotIndex + 1);
```
Comment (ByMartrixx) [inline enigma/src/main/java/cuchaz/enigma/utils/Utils.java:56]: I don't think this handles files or directories starting with a dot and without an extension

**Review by ByMartrixx — CHANGES_REQUESTED** [review]
Just a couple things:
- the tests don't run with `gradlew test`.
- A lot of changes were just 5 imports from the same package changed to a `*` import, which is quite unnecessary imo.

Looks good aside from that and the comments on the files.

**Comment by kadir1243** [comment]
I am going to split this pull(because its a little big) to little pulls

**Pull request closed without merging**